x-pack/test/functional/apps/machine_learning/feature_controls/ml_spaces.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import expect from 'expect.js';
+import { SpacesService } from 'x-pack/test/common/services';
+import { KibanaFunctionalTestDefaultProviders } from '../../../../types/providers';
+
+// tslint:disable:no-default-export
+export default function({ getPageObjects, getService }: KibanaFunctionalTestDefaultProviders) {
+  const esArchiver = getService('esArchiver');
+  const spacesService: SpacesService = getService('spaces');
+  const PageObjects = getPageObjects(['common', 'dashboard', 'security', 'error']);
+  const appsMenu = getService('appsMenu');
+  const testSubjects = getService('testSubjects');
+
+  describe('spaces', () => {
+    before(async () => {
+      await esArchiver.load('empty_kibana');
+    });
+
+    after(async () => {
+      await esArchiver.unload('empty_kibana');
+    });
+
+    describe('space with no features disabled', () => {
+      before(async () => {
+        await spacesService.create({
+          id: 'custom_space',
+          name: 'custom_space',
+          disabledFeatures: [],
+        });
+      });
+
+      after(async () => {
+        await spacesService.delete('custom_space');
+      });
+
+      it('shows Machine Learning navlink', async () => {
+        await PageObjects.common.navigateToApp('home', {
+          basePath: '/s/custom_space',
+        });
+        const navLinks = (await appsMenu.readLinks()).map(
+          (link: Record<string, string>) => link.text
+        );
+        expect(navLinks).to.contain('Machine Learning');
+      });
+
+      it(`can navigate to app`, async () => {
+        await PageObjects.common.navigateToApp('ml', {
+          basePath: '/s/custom_space',
+        });
+
+        await testSubjects.existOrFail('ml-jobs-list', 10000);
```

Comment:
nit: is the custom timeout necessary here?

Reply:
Nope, just copied code... will remove.